In [171]:
import nltk
nltk.download('vader_lexicon')

[nltk_data] Downloading package vader_lexicon to C:\Users\Kaden's
[nltk_data]     Laptop\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [172]:
!pip install translate-api
!pip install translators --upgrade

Defaulting to user installation because normal site-packages is not writeable
Defaulting to user installation because normal site-packages is not writeable


In [173]:
import translators as ts

In [174]:
import string
%matplotlib inline
import pandas as pd
from nltk import sentiment
from IPython.display import display, Markdown
from nltk.sentiment.vader import SentimentIntensityAnalyzer
from nltk import tokenize
from typing import *

# Sentiment_Analyzer

#### BOTH OTHER FILES STORED IN THIS REPOSITORY WERE USED AS REFERENCE FOR THIS FILES CREATION

We begin by importing the data file and looking at a portion of our data

In [175]:
data = pd.read_csv(r"C:\Users\Kaden's Laptop\Documents\GitHub\KadenFranklin.github.io\sentiment_analyzer\names1.csv")
print(data.head())
data.describe()

  Local Names                   Translation    Location Real Names
0      Abombo                             _  Kisanagani     Dupont
1       Adami                         Madam      Opala     generic
2     Adamisu  Knowing eyes, control, anger      Basoko          _
3       Aduyi                          Lion           _          _
4       Aginu                      Kneeling   Kisangani      Genot


Local Names Translation Location Real Names
count          271         271      267        271
unique         270         180       81         99
top      Tuku-Tuku           _  Katanga          _
freq             2          68       28        130

The data gathered for this analyis is collected from King Leopold's Congo and the "Scramble for Africa" by Michael A. Rutz

Specifically from page 163 and the 'Sample of names used in this word'.

We can see from the commands above that there are 271 rows of names, with 270 unique names, 188 distinct translations, originating from 91 locations. In addition to this, asside from collective 'meta' names there are 99 names given to colonials by natives.

The datatypes are as follows:

In [176]:
data.dtypes

Local Names    object
Translation    object
Location       object
Real Names     object
dtype: object

Local Name - Object

Translation - Object

Location - Object	

Real Name - Object

Now just to be certain we will check the dataset for duplicates.

In [177]:
i = -1
for x in data.duplicated():
    i += 1
    
    if x == True:
        print(x)
        print(i)

Success, no duplicates, Now we move on to the interesting part, [sentiment analysis](https://monkeylearn.com/sentiment-analysis/).

Now we can split our data into its corresponding columns so that we can work with it more easily.


In [178]:
names_1 = list(data['Local Names'])
transl = list(data["Translation"])
location = list(data["Location"])
names_2 = list(data['Real Names'])

This piece of code shows any name with punctuation, it will break when encountering nan or null values and was used for error testing.

In [179]:
str_punk = '''!()-[]{};:'"\,<>./?@#$%^&*~'''
no_p = ""

for item in names_2:
    for el in str_punk:
        if el in item:
            print(item)
            

D'Hondt
Cloetens et al.
G. Schweinfurth
G. Miani
P.-J. Van de Moere
P.-J. Van de Moere
H. Bombeeck
L. Oswald, L. Adhémar, C. de Leuze
L. Oswald, L. Adhémar, C. de Leuze
T.Nicolas
G. Dineur
F. Van Nimmen, J. Van Uden
F. Van Nimmen, J. Van Uden
R. Bearts
A. Demunster
A. Lippens
E. Schnitzer (Emin Pacha)
E. Schnitzer (Emin Pacha)
E. Schnitzer (Emin Pacha)
L. De Brandt
A.J.G. Bolle
J. de Dixmude
A.J.G. Bolle
G. Junker
M. Spelier
G.O. Scioz
P. Ponthier
M. Caetano Pereira
P.J. Van de Moere
Bell, M. Polis
Bell, M. Polis
A. De Meulemeester
Mayo-Monteiro
Bandundu, Bulungu
H. Bombeeck 
M. Caetano Pereira
G. Miani
G. Miani
J.M.F. Tumers
G. Geerearts
H. Dupuis Shaw
J. Fievez
J. Fievez
J. Fievez
G. Casati
L. A. Borms
J. Fievez
J. Fievez
C. Van de Laniotte
V. Rue
N. Tobback
L. Van den Brooke
L. Van den Brooke
J. Vansina


The function below defines the [markdown](https://www.markdownguide.org/) table, which the results of our analysis will be stored.

As you can see there are also some helper function I've created to help iterate through data.

In [180]:
def show_markdown_table(headers: List[str], data: List) -> str:
    s = f"| {' | '.join(headers)} |\n| {' | '.join([(max(1, len(header) - 1)) * '-' + ':' for header in headers])} |\n"
    for row in data:
        s += f"| {' | '.join([str(item) for item in row])} |\n"
    display(Markdown(s))

The function below takes in an input of either one name, or a list of names. It will return a resulting markdown table giving the scores of such input names.

In [181]:
def en_name_analyzer(name):

    sid = SentimentIntensityAnalyzer()    

    return sid.polarity_scores(name)

In [182]:
def sw_name_analyzer(name):
    
    for x, elem in enumerate(name):

        name[x] = ts.google(elem)

    sid = SentimentIntensityAnalyzer()    

    return sid.polarity_scores(name)

In [183]:
    #dis=[]
    #sid = SentimentIntensityAnalyzer()
    
    #if isinstance(name, list):
    #    dis.append([[sid.polarity_scores(x)] for x in name])
    #    return dis
                
    #else:
    #    return sid.polarity_scores(name)

In [ ]:
print(names_1, '\n')
print(sw_name_analyzer(names_1))

# should put some other print statement here.

['Abombo', 'Adami', 'Adamisu', 'Aduyi', 'Aginu', 'Agomiriya', 'Alali na se', 'Alinga Mama', 'Alinga Likaya', 'Alinga Tumba', 'Alube', 'Angbalima', 'Angbotalikume', 'Angwadima', 'Asali Monkanda', 'Atama-Atama', 'Atamu', 'Ataso', 'Atila Okondo', 'Awaya-Awaya', 'Bajunu', 'Bakola', 'Bala-Bala', 'Balikpe', 'Balikuhe', 'Bambenga', 'Bana Muanama', 'Bandungu', 'Bapenda Kula', 'Basikoti', 'Batu Pamba', 'Belei', 'Bikengakenga', 'Bola-Bola', 'Bolemba', 'Bombeki', 'Bosolo', 'Bula Matari', 'Bula Matende', 'Bumba', 'Bwana Kapia', 'Bwana Kaseya', 'Bwana Kenengene', 'Bwana Kidogo', 'Bwana Kioseni', 'Bwana Kitoko', 'Bwana Leke', 'Bwana Lubuku', 'Bwana Lutshina', 'Bwana Maibwe', 'Bwana Mukubwa', 'Bwana Mzuri', 'Bwana Ndeke ', 'Bwana Ndeke', 'Bwana Nioka ', 'Bwana Nzige', 'Bwana Pangabisoto ', 'Bwana Tomasi', 'Bwana Tumbaku', 'Chakundia', 'Chicotte', 'Cibalabala', 'Cibutama', 'Cimpanga', 'Ciswa-bantu', 'Citoko', 'Djeke', 'Djoko deli', 'Ebeniq', 'Ebuka-Buka', 'Efanja', 'Ekanda', 'Ekanga-kanga', 'Ekatankoi

Having defined such functions we can perform an analysis on our own dataset.

The code below is complex, but I will annotate it to the best of my ability.
 
 comments are denoted with '#' marks.

In [ ]:
headers = ['local name', 'score', 'location', 'translation', 'real name']
# This line denotes the first line of output data, our heading

rows = [[names_1[num], sw_name_analyzer(names), location[num], 
         str(transl[num] + " " + str(en_name_analyzer(trnl)['compound'])), 
         str(names_2[num] + " " + str(en_name_analyzer(name_2)['compound']))] 
        for num, (names, name_2, trnl) in enumerate(zip(names_1, names_2, transl))]
# This section of code defines the rows that will be generated.
# It relies on various other functions and lists define above, but this is the most critical section.

# 1st line - local names, local name scores, and location lists
# 2nd line - Translation list
# 3rd line - Real names / origin list
# last line - for #, (variables) in enumerate(three different lists)


show_markdown_table(headers, rows)
# This function combines the two lines above to compile output

# can also do # sw_name_analyzer(names)['compound']

Now that we have performed the necessary evaluation of our names, lets organize the data.

In [ ]:
# Organize by region here

In this file we have used the [Vader](https://towardsdatascience.com/sentimental-analysis-using-vader-a3415fef7664) library to perform Sentiment Analysis on a list of roughly 270 names. Vader allows for the performance of multilingual Sentiment analysis through 3rd party translation APIs. What this means is that inputs from other languages are translated to english in order to calculate a sentiment score. I tried looking for a Natural Language Processing database in Swahili, Kikongo, & Lingala. But resources in computer science are sparse in these langauges.

For the reasons above, the words with translated sentiment should be weighted less in the consideration of the word's lasting impact today.